src/CoreShop/Component/Core/Customer/CustomerTransformHelper.php

```diff
@@ -126,7 +126,7 @@ public function moveAddressToNewAddressStack(AddressInterface $address, ElementI
         }
 
         // set new or changed parent
-        if ($this->isNewEntity($address) === true || $address->getParent()->getId() !== $newParent->getId()) {
+        if ($this->isNewEntity($address) === true || ($address->getParent() && $address->getParent()->getId() !== $newParent->getId())) {
```

Comment:
```suggestion
        if ($this->isNewEntity($address) === true || $address->getParent()?->getId() !== $newParent->getId()) {
```
We could simplify with a null safe operator, WDYT? :

Reply:
@ramundomario looks good to me, updated.